recipes/config.rb

```diff
@@ -1,21 +1,60 @@
-template '/etc/default/twemproxy.override' do
-  source 'default.erb'
-  mode '0644'
-  owner 'root'
-  group 'root'
-  variables :args => "--conf-file=#{node['twemproxy']['config_file']}"
-  notifies :reload, 'service[twemproxy]'
-end
+#
+# Cookbook Name:: twemproxy
+# Recipe:: config
+#
+# Copyright 2014, Virender Khatri
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-directory ::File.dirname(node['twemproxy']['config_file']) do
-  recursive true
+template node['twemproxy']['service_config_file'] do
```

Comment:
`--conf-file=#{node['twemproxy']['config_file']}` should be hard coded because it's part of internal chef cookbook behaviour. But `args` should be something more, for specific options.
it can be `node['twemproxy']['launch_options']` or something like that and can be override in any role/environment.